solc/CommandLineInterface.cpp

```diff
@@ -499,13 +515,38 @@ bool CommandLineInterface::processInput()
 
 	function<pair<string,string>(string const&)> fileReader = [this](string const& _path)
 	{
-		auto path = boost::filesystem::path(_path);
-		if (!boost::filesystem::exists(path))
+		// Try to find the longest prefix match in all remappings. At the end, there will be an
+		// empty remapping so that we also try the path itself.
+		int errorLevel = 0;
+		size_t longestPrefix = 0;
+		string bestMatchPath;
+		for (auto const& redir: m_remappings)
+		{
+			auto const& virt = redir.first;
+			if (longestPrefix > 0 && virt.length() <= longestPrefix)
+				continue;
+			if (virt.length() > _path.length() || !std::equal(virt.begin(), virt.end(), _path.begin()))
+				continue;
+			string path = redir.second;
+			path.append(_path.begin() + virt.length(), _path.end());
```

Comment:
Looks like if the user forgets to specify the ending slash in the parameter, we won't find the correct file, right? Should we detect whether user specify that ending slash?